test/Riok.Mapperly.Tests/Mapping/StringToEnumExplicitMapTest.cs

```diff
@@ -0,0 +1,127 @@
+            "public enum E {A = 100, B, C, d, e, E, f}"
+        );
+        TestHelper
+            .GenerateMapper(source, TestHelperOptions.AllowDiagnostics)
+            .Should()
+            .HaveSingleMethodBody(
+                """
+                return source switch
+                {
+                    nameof(global::E.A) => global::E.A,
+                    nameof(global::E.B) => global::E.B,
+                    nameof(global::E.C) => global::E.C,
+                    nameof(global::E.d) => global::E.d,
+                    "str-e" => global::E.e,
+                    nameof(global::E.E) => global::E.E,
+                    nameof(global::E.f) => global::E.f,
+                    _ => System.Enum.Parse<global::E>(source, false),
+                };
+                """
+            )
+            .HaveAssertedAllDiagnostics();
+    }
+
+    [Fact]
+    public void EnumFromStringWithExplicitValueMultipleSourcesToOneEnum()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "[MapEnumValue(\"str-e1\", E.e), MapEnumValue(\"str-e2\", E.e)] public partial E FromStr(string source);",
+            "public enum E {A = 100, B, C, d, e, E, f}"
+        );
+        TestHelper
+            .GenerateMapper(source, TestHelperOptions.AllowDiagnostics)
+            .Should()
+            .HaveSingleMethodBody(
+                """
+                return source switch
+                {
+                    nameof(global::E.A) => global::E.A,
+                    nameof(global::E.B) => global::E.B,
+                    nameof(global::E.C) => global::E.C,
+                    nameof(global::E.d) => global::E.d,
+                    "str-e1" => global::E.e,
+                    "str-e2" => global::E.e,
+                    nameof(global::E.E) => global::E.E,
+                    nameof(global::E.f) => global::E.f,
+                    _ => System.Enum.Parse<global::E>(source, false),
+                };
+                """
+            )
+            .HaveAssertedAllDiagnostics();
+    }
+
+    [Fact]
+    public void EnumToStringWithExplicitValueDuplicatedSource()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "[MapEnumValue(\"str-e\", E.e), MapEnumValue(\"str-e\", E.E)] public partial E FromStr(string source);",
+            "public enum E {A = 100, B, C, d, e, E, f}"
+        );
+        TestHelper
+            .GenerateMapper(source, TestHelperOptions.AllowDiagnostics)
+            .Should()
+            .HaveSingleMethodBody(
+                """
+                return source switch
+                {
+                    nameof(global::E.A) => global::E.A,
+                    nameof(global::E.B) => global::E.B,
+                    nameof(global::E.C) => global::E.C,
+                    nameof(global::E.d) => global::E.d,
+                    "str-e" => global::E.e,
+                    "str-e" => global::E.E,
```

Comment:
Shouldn't the generated code in this case only include one unique case and the second one should be ignored as it is not reachable anyway.
(Should be as easy as adding a `continue;` in `MappingBuilders.StringToEnumMappingBuilder.BuildExplicitValueMappings if (!checkedSources.Add(source.ConstantValue.Value))`

Reply:
Yes, I agree. I have added continue.